lib/sepa.js

```diff
@@ -616,10 +619,28 @@
         r(mandate, 'MndtId', this.mandateId);
         r(mandate, 'DtOfSgntr', this.mandateSignatureDate.toISOString().substr(0, 10));
 
-        if (this.ammendment) {
+        // Add amendment information - always include AmdmntInd to indicate amendment status
+        if (this.amendment && this.amendment.isAmended) {
```

Comment:
Are there cases where the amendment object is specified but isAmended is false? I'm wondering if it makes sense to simplify this and to a single property with the original creditor scheme id.

Then again your error message suggests there might be cases where there are other amendment details other than originalCreditorSchemeId.

I still think we could drop isAmended. If `amendment` is not specified, you fall into the else case and set AmdmntInd to false. If it is specified you check for the `originalCreditorSchemeId` and set it accordingly, or bail with the error message if no details.

Does that make sense in practice?

Reply:
Sounds good!